test/stmt/strict_exhaustive_switches.swift

```diff
@@ -0,0 +1,113 @@
+// RUN: %target-typecheck-verify-swift -swift-version 6 -enable-experimental-feature StrictExhaustiveSwitches
+// REQUIRES: swift_feature_StrictExhaustiveSwitches
+
+enum ReasonableEnum {
+  case zero
+  case one
+}
+
+enum OverlyLargeSpaceEnum {
+  case case0
+  case case1
+  case case2
+  case case3
+  case case4
+  case case5
+  case case6
+  case case7
+  case case8
+  case case9
+  case case10
+  case case11
+}
+
+func testSwitch() -> Bool {
+  // Single subject switch
+
+  // well-formed exhaustive enum
+  switch ReasonableEnum.zero {
+    case .zero, .one: return true
+  }
+
+  // malformed exhaustive plus default enum will warn in silgen
+  switch ReasonableEnum.zero {
+    case .zero, .one: return true
+    default: return false
+  }
+
+  switch OverlyLargeSpaceEnum.case1 {
+    // expected-note@-1 {{add missing cases}}
+    // expected-note@-2 11 {{add missing case}}
+    case .case0: return true
+    default: return false // expected-warning {{switch has known exhaustive cases, remove default case}}
+  }
+
+  // Tuple cases
+
+  switch (ReasonableEnum.zero, ReasonableEnum.one) { // expected-note {{add missing cases}}
+    // expected-note@-1 {{add missing case: '(.one, _)'}}
+    // expected-note@-2 {{add missing case: '(_, .one)'}}
+    case (.zero, .zero): return true
+    default: return false // expected-warning {{switch has known exhaustive cases, remove default case}}
+  }
+
+  switch (OverlyLargeSpaceEnum.case1, OverlyLargeSpaceEnum.case2) { // expected-note {{add missing case: '(.case11, _)'}}
+  case (.case0, _): return true
+  case (.case1, _): return true
+  case (.case2, _): return true
+  case (.case3, _): return true
+  case (.case4, _): return true
+  case (.case5, _): return true
+  case (.case6, _): return true
+  case (.case7, _): return true
+  case (.case8, _): return true
+  case (.case9, _): return true
+  case (.case10, _): return true
+  default: return false // expected-warning {{switch has known exhaustive cases, remove default case}}
+  }
+
+  // Open type case
+
+  // exhaustive type-checks correctly
+  let number = 2
+  switch number {
+  case 1: return true
+  case 2...10: return true
+  default: return false
+  }
+
+  // nonexhaustive warns as expected
+  switch number { // expected-error {{switch must be exhaustive}}
+  // expected-note@-1 {{add a default clause}}
+  case 1: return true
+  case 2...10: return true
+  }
+
+  // Unkonwn default
```

Comment:
```suggestion
  // Unknown default
```

The one `@unknown` default case I'd like to see handled right off the bat (IMO), perhaps even without the new compiler option, is unambiguously unnecessary use of non-`@unknown` default when an `@unknown` default will do. That is, if all known cases of a resilient enum are already handled, it should be at least a warning to unwittingly defeat future diagnostics by writing plain `default`. (This may already be diagnosed--I'm not sure.)